service-base/src/main/java/org/eclipse/hono/service/http/HttpServiceBase.java

```diff
@@ -441,21 +420,24 @@ protected HttpServerOptions getInsecureHttpServerOptions() {
         return Future.succeededFuture();
     }
 
-    private void doGetStatus(final RoutingContext ctx) {
-        JsonObject result = new JsonObject();
-        result.put("active profiles", activeProfiles);
-        adaptStatusResource(result);
-        ctx.response()
-                .putHeader(HttpHeaders.CONTENT_TYPE, CONTENT_TYPE_JSON)
-                .end(result.encodePrettily());
-    }
+    /**
+     * Register readiness checks for the service.
+     * <p>
+     * Subclasses must override this method to register service specific checks.
+     *
+     * @param handler The health check handler to register the checks with.
+     */
+    @Override
+    public abstract void registerReadinessChecks(HealthCheckHandler handler);
```

Comment:
It promotes the usage of health check. Do you have a meaningful default implementation for `HttpServiceBase` in mind? If we have none, I would suggest to remove the methods for the health check entirely from `HttpServiceBase`, otherwise we would override an empty implementation from the parent class (`AbstractServiceBase`) by an empty implementation, which would be IMHO useless.